samcli/lib/list/endpoints/endpoints_producer.py

```diff
@@ -0,0 +1,472 @@
+"""
+The producer for the 'sam list endpoints' command
+"""
+import dataclasses
+import logging
+from typing import Dict, List, Any
+from enum import Enum
+import json
+from botocore.exceptions import ClientError, BotoCoreError
+from samcli.commands.list.exceptions import (
+    SamListUnknownBotoCoreError,
+    SamListLocalResourcesNotFoundError,
+    SamListUnknownClientError,
+)
+from samcli.lib.list.list_interfaces import Producer
+from samcli.lib.providers.sam_stack_provider import SamLocalStackProvider
+from samcli.lib.providers.provider import Stack
+from samcli.commands._utils.template import get_template_data
+from samcli.lib.list.endpoints.endpoints_def import EndpointsDef
+from samcli.lib.list.resources.resource_mapping_producer import ResourceMappingProducer
+from samcli.lib.utils.boto_utils import get_client_error_code
+
+LOG = logging.getLogger(__name__)
+ENDPOINT_RESOURCE_TYPES = {"AWS::Lambda::Function", "AWS::ApiGateway::RestApi", "AWS::ApiGatewayV2::Api"}
```

Comment:
We can use the static vars in here: https://github.dev/aws/aws-sam-cli/blob/e2a6e865493bee10b4aa3258a84891f2650fe009/samcli/lib/utils/resources.py#L11 (and same goes with rest of resource type strings)